### ISSUE TITLE
Add fancy input APIs

### ISSUE THREAD

**Issue by paulirish** [puppeteer/puppeteer#40]
```js
.mouseMoved(x, y, options = {})
.mousePressed(x, y, options = {})
.mouseReleased(x, y, options = {})
.tap(x, y, options = {})
.touchmove()
.touchend()
```

**Comment by JoelEinbinder** [comment]
For tapping:
`Input.synthesizeTapGesture` appears to only work on android
`Input.dispatchTouchEvent` doesn't click events like normal tapping would
`Input.emulateTouchFromMouseEvent` fires mouse events, but only after an undefined delay.

The best way to emulate touch right now appears to be setting the device mode to iPhone, which sends touchStart and touchEnd along with every mouse press.

**Comment by JoelEinbinder** [comment]
Closing this in favor of smaller issues

#156 
#157 
#158
